base/inference.jl

```diff
@@ -52,20 +52,39 @@ const NF = NotFound()
 typealias LineNum Int
 typealias VarTable Array{Any,1}
 
+# The type of a variable load is either a value or an UndefVarError
 type VarState
     typ
     undef::Bool
     VarState(typ::ANY, undef::Bool) = new(typ, undef)
 end
 
-immutable Const
+# The type of a value might be constant
+type Const
```

Comment:
I don't think we should change this as a performance hack for a compiler feature that doesn't exist yet. The primary concern is the semantics --- if it's intended to be immutable, we should feel free to declare it immutable.